packages/react-dom/src/events/ChangeEventPlugin.js

```diff
@@ -77,31 +74,6 @@ function shouldUseChangeEvent(elem) {
   );
 }
 
-function manualDispatchChangeEvent(nativeEvent) {
-  const event = createAndAccumulateChangeEvent(
-    activeElementInst,
-    nativeEvent,
-    getEventTarget(nativeEvent),
-  );
-
-  // If change and propertychange bubbled, we'd just bind to it like all the
-  // other events and have it go through ReactBrowserEventEmitter. Since it
-  // doesn't, we manually listen for the events and so we have to enqueue and
-  // process the abstract event manually.
-  //
-  // Batching is necessary here in order to ensure that all event handlers run
-  // before the next rerender (including event handlers attached to ancestor
-  // elements instead of directly on the input). Without this, controlled
-  // components don't work properly in conjunction with event bubbling because
-  // the component is rerendered and the value reverted before all the event
-  // handlers can run. See https://github.com/facebook/react/issues/708.
-  batchedUpdates(runEventInBatch, event);
```

Comment:
This code dispatches a change event; it doesn't do anything by itself. Can we figure out why this was breaking?

Reply:
I don't really feel comfortable removing this logic until we understand why it's buggy.

Reply:
Totally, 👍

Reply:
As far as I can tell, this code executes correctly. I believe the issue is a difference in browser behavior when using `attachEvent` vs `addEventListener`. The following code snippet seems to demonstrate what is happening:

Try this in IE9:
https://s.codepen.io/nhunzaker/debug/Lrpqov/ZoABaKanBoKr

Browse the code:
https://codepen.io/nhunzaker/pen/Lrpqov

It looks like IE9, when using `attachEvent` for text input, causes previously referenced text nodes to be replaced with a copy. This behavior doesn't present itself if you use `addEventListener`.

I think we were able to get away with this before (but I'm not super familiar) because we identified text via an embedded span, and then eventually relative to a comment.

Reply:
This is... probably the strangest DOM thing I have seen. I would love a second/third opinion here.

Reply:
Dang. Here is what I am seeing...

On every `onpropertychange` call, IE9 seems to be marking all text nodes changed within the callback as "dirty". On undo, it replaces those text nodes with copies (though it doesn't seem to revert them)

Additionally, undoing doesn't appear to trigger `onpropertychange`. We just see that in React because it triggers a keyup event.

Reply:
this is soooo special.

The original point of of `onpropertychange` here was specifically for ie8, not ie9. It makes up the bulk of the polyfill, and should be the thing reporting the change. We avoid the ie9 oninput because it's a bit buggy, however without needing IE8 support we can craft a _leaner_ polyfill that uses the selectionchange stuff as well as the native onInput event.

I don't think it's safe to remove the `onpropertychange` tho unless it's backfilled with the `onInput` event, which (I think) is skipped for ie9.

Reply:
> without needing IE8 support we can craft a leaner polyfill that uses the selectionchange stuff as well as the native onInput event.

I think this is already happening. If we remove `onpropertychange`, the combination of keyboard events and selection change, so far, appears to cover this use case.

Reply:
I'm seeing a combination of input, keyup, keydown, and selection change:

![jun-05-2018 09-41-26](https://user-images.githubusercontent.com/590904/40979756-b8aa38a4-68a4-11e8-93e7-0a94705b27ae.gif)

All of the extra calls get filtered out via `inputValueTracking`.

Reply:
really huh? That seems weird the input support check specifically checks for `documentMode > 9`

Reply:
Looks like I'm wrong. I was looking directly what what was getting dispatched out of [ReactDOMEventListener](https://github.com/facebook/react/blob/master/packages/react-dom/src/events/ReactDOMEventListener.js).

I did find one edge case that propertychange catches:

1. Change text
2. Right clickSwitch to another input

Reply:
wouldn't the onInput event catch that as well? Or does it not fire in that case for ie9?

Reply:
According to MDN:

> IE 9 does not fire an input event when the user deletes characters from an input (e.g. by pressing Backspace or Delete, or using the "Cut" operation).

https://developer.mozilla.org/en-US/docs/Web/Events/input (footnote 2)

Reply:
right, deleting doesn't work, we have the `selectionchange` and various key events to catch those cases tho (that is the intent anyway)

Reply:
Right, right I thought you were asking about `input` events with `onInput`, but you were probably talking about our polyfilled version :)

Reply:
I wonder if we could catch _some_ of these edge cases by comparing the value on focus:

https://github.com/facebook/react/blob/master/packages/react-dom/src/events/ChangeEventPlugin.js#L135-L139

I'm not sure what to do about "Cut", though I can't help but wonder if it triggers a selection change (you have text selected, you cut, and thus, the words disappear and selection changes).